### PR TITLE
provide support for ShowMode in SwitchTo and Cancel widgets #350

### PR DESCRIPTION
**Pull request by lubaskinc0de**
Close issue #350, now users can pass show_mode argument to Cancel and SwitchTo widgets

```py
third_window = Window(
    "Third Window",
    Cancel(Const("Back"), show_mode=ShowMode.DELETE_AND_SEND),
    state=SecondSG.main,
)

second_dialog = Dialog(third_window)

second_window = Window(
    "Second",
    SwitchTo(Const("To first"), state=MySG.first, id="to_first"),
    state=MySG.second,
)

first_window = Window(
    "First",
    SwitchTo(Const("To second"), state=MySG.second, id="to_second", show_mode=ShowMode.SEND),
    Start(Const("To next dialog"), state=SecondSG.main, id="to_next_dialog"),
    state=MySG.first,
)
first_dialog = Dialog(first_window, second_window)
```

### CODE REVIEW

File: src/aiogram_dialog/widgets/kbd/state.py [hunk 1]
```diff
@@ -41,6 +43,7 @@ def __init__(
             state: State,
             on_click: Optional[OnClick] = None,
             when: WhenCondition = None,
+            show_mode: ShowMode = None,
```
Comment (Tishka17) [inline src/aiogram_dialog/widgets/kbd/state.py:46]: `None` is invalid value for type hint `ShowMode`

Comment (lubaskinc0de) [reply src/aiogram_dialog/widgets/kbd/state.py]: fixed

**Comment by lubaskinc0de** [comment]
should we use ShowMode.AUTO for default or make it Optional?

File: src/aiogram_dialog/widgets/kbd/state.py [hunk 2]
```diff
@@ -117,6 +121,7 @@ def __init__(
             result: Any = None,
             on_click: Optional[OnClick] = None,
             when: WhenCondition = None,
+            show_mode: ShowMode = ShowMode.AUTO,
```
Comment (Tishka17) [inline src/aiogram_dialog/widgets/kbd/state.py:124]: let's do it in the same way as in manager methods:
```py
show_mode: Optional[ShowMode] = None,
```

Comment (lubaskinc0de) [reply src/aiogram_dialog/widgets/kbd/state.py]: done

File: src/aiogram_dialog/widgets/kbd/state.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 from aiogram_dialog.widgets.common import WhenCondition
 from aiogram_dialog.widgets.text import Const, Text
 from aiogram_dialog.widgets.widget_event import WidgetEventProcessor
+from aiogram_dialog import ShowMode
```
Comment (Tishka17) [inline src/aiogram_dialog/widgets/kbd/state.py:11]: let's have full imports from api package like above

Comment (lubaskinc0de) [reply src/aiogram_dialog/widgets/kbd/state.py]: done

**Pull request merged**